### PR TITLE
allow setting remote file metadata

### PR DESCRIPTION
**Pull request by verdude**
This should fix viewing resources in the ayamel player.

Needed to save the changes into `resource.content.files` and make sure there was a `mime` field which is the same as the mimeType field AFAICT.

### CODE REVIEW

File: src/resources/index.ts [hunk 1]
```diff
@@ -19,33 +24,49 @@ export default async function(app: Express) {
...
+      if (!file?.mime && file?.mimeType) {
+        // copy mimetype
+        // The ayamel player requires mime to be set but
+        // the backend does not set a mime field.
+        file.mime = file.mimeType;
+      }
+      return file;
+    });
+
+    Resource.findByIdAndUpdate(req.params.id, { $set: { "content.files": files } }, { new: true }, (err, resource) => {
```
Comment (verdude) [inline src/resources/index.ts:58]: Not sure if this is the best way to update docs in mongo. We are replacing the entire files array, I don't know if that will be an issue elsewhere.

File: src/resources/index.ts [hunk 2]
```diff
@@ -11,6 +11,11 @@ const Relation = mongoose.model('relations', relationSchema);
 
 export default async function(app: Express) {
 
+  function fixId(doc: any) {
```
Comment (verdude) [inline src/resources/index.ts:14]: I'm fairly new to express, is there a better place to store utility functions?

Comment (gliese1337) [reply src/resources/index.ts:14]: Not really, this is fine.

File: src/resources/index.ts (hunk 1 above)
Comment (gliese1337) [reply src/resources/index.ts:58]: That should be fine for now. I think that was the behavior of the original resource library, but if it does cause problems we can try to figure out how to do a proper merge later.

**Review by gliese1337 — APPROVED** [review]

File: src/resources/index.ts (hunk 2 above)
Comment (ansorensen) [reply src/resources/index.ts:14]: Sometimes I aggregate them elsewhere in utils file or directory and import them, but this is fine.

**Pull request merged**